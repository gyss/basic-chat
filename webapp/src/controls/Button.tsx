/** @jsx jsx */

import * as React from 'react'
import {css, jsx} from '@emotion/core'
import styled from '@emotion/styled'

const StyledButton = styled.button`
  background-color: var(--primary-color);
  border-color: transparent;
  color: #fff;
  cursor: pointer;
  font-size: 1.125rem;
  padding-left: 1.125rem;
  padding-right: 1.125rem;
  ${(props: IProps) => props.fullWidth && 'width: 100%;'}
  &:focus {
    outline: none;
  }
  &:hover {
    background-color: var(--secondary-color);
  }
`

interface IProps {
  fullWidth?: boolean
  children: React.ReactNode
}

export default function Button({
  fullWidth,
  children,
  ...props
}: IProps & React.ButtonHTMLAttributes<HTMLButtonElement>) {
  return (
    <StyledButton fullWidth={fullWidth} {...props}>
      {children}
    </StyledButton>
  )
}
