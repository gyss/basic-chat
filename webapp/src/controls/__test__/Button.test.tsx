/** @format */

import * as React from 'react'
import {render, fireEvent, cleanup} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Button from '../Button'

describe('<Button>', () => {
  afterEach(cleanup)

  test('should render', () => {
    const {asFragment} = render(<Button>This is a test</Button>)
    expect(asFragment()).toMatchSnapshot()
  })
})
