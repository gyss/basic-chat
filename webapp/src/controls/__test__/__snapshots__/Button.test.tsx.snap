// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> should render 1`] = `
<DocumentFragment>
  <a
    class="css-8nk2fv"
  >
    This is a test
  </a>
</DocumentFragment>
`;
